composables/useFilterState.ts

```diff
@@ -1,34 +1,53 @@
 import { debouncedRef } from '@vueuse/core';
 
+export type FilterStateOptions<T> = {
+  initialFilters?: T;
+  debounceTimeMs?: number;
+  localStorageKey?: string;
+};
+
 export function useFilterState<T extends Record<string, any>>(
-  initialFilters: T,
-  debounceTimeMs = 300
+  options?: FilterStateOptions<T>
 ) {
-  const filter = ref({ ...initialFilters }) as Ref<T>;
+  const filter = ref<T>({});
+
+  function setFilter<T>(filterToSet) {
+    filter.value = filterToSet;
+    if (options?.localStorageKey) {
+      localStorage.setItem(
+        options.localStorageKey,
+        JSON.stringify(filterToSet)
+      );
+    }
+  }
+
+  if (options?.initialFilters) {
+    setFilter(options.initialFilters);
+  }
 
-  const enabeledFiltersCount = computed(() => {
+  const enabledFiltersCount = computed(() => {
     return Object.values(filter.value).filter(
       (value) => value !== undefined && value !== ''
     ).length;
   });
 
-  const canClearFilter = computed(() => enabeledFiltersCount.value > 0);
+  const canClearFilter = computed(() => enabledFiltersCount.value > 0);
 
   function clear() {
-    filter.value = { ...initialFilters };
+    setFilter({});
```

Comment:
If we're initialising with what's on local storage, then clear should literally clear all filters (clear is not actually used anywhere currently)